sysdeps/sigma/rtdl-generic/support.cpp

```diff
@@ -0,0 +1,105 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+#include <frg/vector.hpp>
+#include <frg/string.hpp>
+#include <frg/eternal.hpp>
+
+#include <mlibc/allocator.hpp>
+#include <mlibc/sysdeps.hpp>
+
+#include <libsigma/thread.h>
+#include <libsigma/file.h>
+#include <libsigma/memory.h>
+
+namespace {
+    class id_allocator{
+        public:
+        constexpr id_allocator(): id(0) {}
+
+        uint64_t get_id(){
+            return id++;
+        } 
+
+        private:
+        uint64_t id;
+    };
+
+    auto file_descriptor_allocator = id_allocator();
+
+    struct file_descriptor {
+        file_descriptor(): fd(0), offset(0), name(frg::string<MemoryAllocator>(getAllocator())) {}
+        file_descriptor(const char* name): fd(file_descriptor_allocator.get_id()), offset(0), name(frg::string<MemoryAllocator>(getAllocator(), name) + '\0') {}
```

Comment:
This coding style is inconsistent with mlibc. Put `: <initialization list>` onto the next line, at the same level of indentation as the constructor declaration.